### DISCUSSION TITLE
brew doctor error

### DISCUSSION THREAD

**Discussion by achzhou** [Homebrew/discussions#3118] [Getting started] [answered]
I had brew installed successfully a long time ago. I haven't used it for a whille. When I tried to use it again, I ran the 'brew doctor' first then I found the following message:
$ brew doctor
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require': cannot load such file -- active_support/core_ext/object/blank (LoadError)
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /usr/local/Homebrew/Library/Homebrew/global.rb:10:in `<top (required)>'
	from /usr/local/Homebrew/Library/Homebrew/brew.rb:13:in `require_relative'
	from /usr/local/Homebrew/Library/Homebrew/brew.rb:13:in `<main>'

I googled it and got some suggestions. Some said that I can tried 'brew update-reset' then 'brew upgrade'. But it didn't work. 
$ brew update-reset
==> Fetching /usr/local/Homebrew...
remote: Enumerating objects: 89163, done.
remote: Counting objects: 100% (12008/12008), done.
remote: Compressing objects: 100% (36/36), done.
error: 5561 bytes of body are still expected MiB | 13.00 KiB/s 
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF
fatal: fetch-pack: invalid index-pack output

==> Resetting /usr/local/Homebrew...
Branch 'master' set up to track remote branch 'master' from 'origin'.
Reset branch 'master'
Your branch is up to date with 'origin/master'.

==> Fetching /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core...
remote: Enumerating objects: 616630, done.
remote: Counting objects: 100% (616630/616630), done.
remote: Compressing objects: 100% (166753/166753), done.
error: 4322 bytes of body are still expected9 MiB | 28.00 KiB/s 
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF
fatal: fetch-pack: invalid index-pack output

==> Resetting /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core...
Branch 'master' set up to track remote branch 'master' from 'origin'.
Reset branch 'master'
Your branch is up to date with 'origin/master'.

So have to start a post here to get some help. Thanks a lot!

**Comment by gromgit** [comment]
> ```
> $ brew update-reset
> ==> Fetching /usr/local/Homebrew...
> remote: Enumerating objects: 89163, done.
> remote: Counting objects: 100% (12008/12008), done.
> remote: Compressing objects: 100% (36/36), done.
> error: 5561 bytes of body are still expected MiB | 13.00 KiB/s
> fetch-pack: unexpected disconnect while reading sideband packet
> fatal: early EOF
> fatal: fetch-pack: invalid index-pack output
> ```
Something's seriously wrong if you're having problems fetching the relatively small Homebrew software repo.

Are you using a proxy, a local firewall, or other security software? If so, can you disable it and try `brew update-reset` again?

**Reply by achzhou** [reply]
Yes, I was using proxy since I am in China now. It looks like that it would be very difficult to access to github without proxy. But if I disabled the proxy and rerun 'brew update-reset', here is what I got:

$ brew update-reset
==> Fetching /usr/local/Homebrew...
fatal: unable to access 'https://github.com/Homebrew/brew/': LibreSSL SSL_connect: Operation timed out in connection to github.com:443 

==> Resetting /usr/local/Homebrew...
Branch 'master' set up to track remote branch 'master' from 'origin'.
Reset branch 'master'
Your branch is up to date with 'origin/master'.

==> Fetching /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core...
fatal: unable to access 'https://github.com/Homebrew/homebrew-core/': Failed to connect to github.com port 443: Operation timed out

==> Resetting /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core...
Branch 'master' set up to track remote branch 'master' from 'origin'.
Reset branch 'master'
Your branch is up to date with 'origin/master'.

Any other suggestions? Thanks a lot!

**Reply by SMillerDev** [reply]
If you're in China I would suggest some local university mirrors

**Comment by achzhou** [comment, marked as answer]
Problem solved. I searched in the Internet and got an answer that I need to rewrite 
/etc/hosts 
file in order to associate ip addresses to gitbub.com so that when my 'brew update-reset' was looking for github.com, the connection won't be blocked. 
Then I rerun 
$ brew update-reset'.
Need to be a little patient to wait for the whole process to be completely finished. There should not be any error message. Then run 
$ brew doctor
then should find out everything should be ok by then.
